### ISSUE TITLE
global-cache-cache-override

### ISSUE THREAD

**Issue by hhaddouch** [wmo-im/wis2-global-services-testing#7]
Where a Global Cache determines that it is unable to cache a data item, the Global Cache should publish a notification message where the data download link refers to the source dataserver and the metric wmo_wis2_gc_cache_override_total is incremented by 1
More details for global-cache-cache-override are needed about the notification message; format, content, topic...

**Comment by golfvert** [comment]
I guess this metrics can be used when a GC decides not to cache something that should be cached. Core data with cache=true or missing directive. In this case, the notification message will be exactly the same it would have been if the original notification message had cache=false.
Not sure anything else is needed.

**Comment by kaiwirt** [comment]
Do we want a GC in this case to republish a message with the original link? My arguments against would be

- The data should be cached. Which means that the originator might not want to share the data by itself
- This contrasts cache=false where the data originator clearly states it wants to share the data by itself, so we have a different situation
- The data might be available by other caches. So not resending the message would allow users to fetch the data from other caches which might be better than accessing the data source

How do we cope with situations where all Caches decide that they can not store the data?

**Comment by golfvert** [comment]
I'd view this from a user point of view... so republish as if cache=false and generate a monitoring message so that the producing centre is aware of this. 
If we do something else, I'm afraid this will have an impact on users. 
Like that it remains a "technicality" between GC and data producers.

**Comment by kaiwirt** [comment]
I am not sure it is that easy. Consider for example a data publisher only allows downloads from the GCs. In this case a cache that is not able to cache that data sends out a message with a link that does not work for the user. Instead of just not sending out a message. (The user will receive messages from other caches with links that work).

I think cache override is more related to a download error than to cache = false

**Comment by kaiwirt** [comment]
Yes, it is a decision from the GC. My point is that the outcome of this decision is closer to an error (i.e. the originator wants the data to be in the cache which is however not possible) than it is to cache = false (the originator does not want the cache to store the data).

My point is that to the user the consequences of receiving a message from that cache in this situation might be worse (the download link provided might not work) than to not receive that message (i.e. download the data from any other cache)

**Comment by jmaxmarno** [comment]
fwiw - i agree with Kai's perspective and I believe it is supported by viewing this from both the Origin, and end user cases. Override scenario:

- the data object was intended (by the Origin) to be cached
- the data object was not cached by the GC

This two characteristics exactly match the download failure case, and the GC should not publish a notification message. 
The only difference between how a download failure scenario, and override scenario is handled is which metric should be incremented.

**Comment by golfvert** [comment]
Fair enough... Optimistically, I thought that overriding and nevertheless providing a notification message was doable as the origin download link may work.

**Comment by 6a6d74** [comment]
This isn't the right repo to discuss functional behaviour. New issue opened to debate. See [Issue et-w2it/4](https://github.com/wmo-im/et-w2it/issues/4). Closing this issue.
